src/net/mod.rs

```diff
@@ -48,8 +63,9 @@ mod conn {
     pub type DomainConnector = Box<dyn TcpDomainConnector>;
 
     /// connect to domain and return connection
-    #[async_trait]
-    pub trait TcpDomainConnector: Send + Sync {
+    #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
+    #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+    pub trait TcpDomainConnector {
```

Comment:
Should probably name this to something like `SocketDomainConnector` but I don't have a great name.